ui/appui-react/src/appui-react/frontstage/FrontstageDef.tsx

```diff
@@ -287,13 +296,24 @@ export class FrontstageDef {
 
     const oldState = this._nineZoneState;
     this._nineZoneState = state;
-    this.handlePopouts(oldState);
+    const popoutResult = this.handlePopouts(oldState);
     if (oldState) {
```

Comment:
I wonder if `onWidgetStateChangedEvent` is emitted when the popout widget fails to open.

Reply:
Looking into further

Reply:
It is emitted

Reply:
With what `WidgetState`? Will we emit a correct `WidgetState` once again when restoring to the "old nine zone" state?

Reply:
It is emitting onWidgetStateChangedEvent whenever it is setting the failed popout back to the old state. It is not emitting the onWIdgetStateChangedEvent with the incorrect state